s3/pkg/service/multipart.go

```diff
@@ -567,3 +568,140 @@ func (s *s3Service) ListObjectParts(ctx context.Context, in *pb.ListObjectPartsR
 
 	return nil
 }
+
+func (s *s3Service) CopyObjPart(ctx context.Context, in *pb.CopyObjPartRequest, out *pb.CopyObjPartResponse) error {
+	log.Info("CopyObjPart is called in s3 service.")
+	var err error
+	defer func() {
+		out.ErrorCode = GetErrCode(err)
+	}()
+
+	srcBucketName := in.SourceBucket
+	srcObjectName := in.SourceObject
+	targetBucketName := in.TargetBucket
+	targetObjectName := in.TargetObject
+	uploadId := in.UploadID
+	partId := in.PartID
+	size := in.ReadLength
+	offset := in.ReadOffset
+
+	srcBucket, err := s.MetaStorage.GetBucket(ctx, srcBucketName, true)
+	if err != nil {
+		log.Errorln("get bucket failed with err:", err)
+		return err
+	}
+	srcObject, err := s.MetaStorage.GetObject(ctx, srcBucketName, srcObjectName, "", true)
+	if err != nil {
+		log.Errorln("failed to get object info from meta storage. err:", err)
+		return err
+	}
+
+	isAdmin, tenantId, _, err := util.GetCredentialFromCtx(ctx)
+	if err != nil {
+		log.Error("get tenant id failed, err:", err)
+		err = ErrInternalError
+		return err
+	}
+	//check source object acl
+	if !isAdmin {
+		switch srcObject.Acl.CannedAcl {
+		case "public-read", "public-read-write":
+			break
+		default:
+			if srcObject.TenantId != tenantId {
+				err = ErrAccessDenied
+				return err
+			}
+		}
+	}
+
+	_, _, _, err = CheckRights(ctx, srcObject.TenantId)
```

Comment:
As acl is already checked, do we need this CheckRights anymore?

Reply:
thanks, will delete the code block